### ISSUE TITLE
Critical: broken on CPython 3.8 & redis 4.2+

### ISSUE THREAD

**Issue by vytas7** [dsoftwareinc/fakeredis-py#3]
**Describe the bug**
The latest (at the time of this writing) release [fakeredis-1.7.4](https://pypi.org/project/fakeredis/1.7.4/) is completely broken on CPython 3.8.

**To Reproduce**
Assuming `docker` is available, simply run 
```
docker run --rm -it python:3.8 /bin/bash -c "pip install fakeredis; python -c 'import fakeredis; fakeredis.FakeRedis()'"

<...>
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/fakeredis/_server.py", line 2804, in __init__
    if not bound.arguments['connection_pool']:
KeyError: 'connection_pool'
```

Interestingly, CPython 3.9 works fine.

**Expected behavior**
Expected to be able to initialize an instance of `fakeredis.FakeRedis`.

**Comment by cunla** [comment]
Hi, this is strange, tests with python3.8 pass (see [here](https://github.com/dsoftwareinc/fakeredis-py/runs/6336667578?check_suite_focus=true)).

**Comment by henu** [comment]
We got this same problem with the new version

**Comment by adam-sikora** [comment]
Look like it is happening for `python` 3.8 and `redis` 4.2+

Same issue as I reported in the old repository: https://github.com/jamesls/fakeredis/pull/330#issuecomment-1103973466

**Comment by Sherry112** [comment]
I am also facing the same issue.

**Comment by cunla** [comment]
Fixed in 1.7.5.
